workers/social/lib/social/models/account.coffee

```diff
@@ -1239,3 +1244,94 @@ module.exports = class JAccount extends jraphical.Module
     options  = { limit, skip, sort }
     JSession = require './session'
     JSession.some selector, options, callback
+
+
+  destroyAccount$: secure (client, password, callback) ->
+
+    { delegate } = client.connection
+    { profile: { nickname } } = delegate
+
+    checkUserPassword delegate, password, (err) =>
+
+      return callback new KodingError err  if err
+
+      @destroyAccount client, callback
+
+
+  destroyAccount: (client, callback) ->
+
+    @fetchRelativeGroups (err, groups) =>
+
+      # remove koding team
+      groups = groups.filter (group) -> group.slug isnt 'koding'
+
+      ownerOfGroups = groups.filter (group) -> 'owner' in group.roles
+
+      if ownerOfGroups.length > 1
+        return callback new KodingError 'You cannot delete your account when you have ownership in other team', ownerOfGroups
+
```

Comment:
again in here if you provide `query` support in `fetchRelativeGroups` you can use it in here too.